### PR TITLE
CVE-2014-7925 & CVE-2011-3880

### PR DESCRIPTION
**Pull request by chikeud**
CVE-2014-7925 & CVE-2011-3880

### CODE REVIEW

File: cves/CVE-2011-3880.yml [hunk 1]
```diff
@@ -32,7 +32,17 @@ description_instructions: |
...
-description: 
+description: Google Chrome before 15.0.874.102 does not neutralize an invalid character (\0) in HTTP headers which can lead to many opportunities
+  for XSS and CRSF by attackers. After receiving and interpreting a request message, a server responds with an HTTP response message.
+  \0 character inside a headerline is being interpreted as a terminator.
+  But \00 can also work
+  An attacker can bypass wep application filters like this -
+
+  $location = str_replace("\r","",$location);
+  $location = str_replace("\n","",$location);
+
```
Comment (msp2248) [inline cves/CVE-2011-3880.yml]: Elaborating more on what that code does would help

File: cves/CVE-2011-3880.yml [hunk 2]
```diff
@@ -32,7 +32,17 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Google Chrome before 15.0.874.102 does not neutralize an invalid character (\0) in HTTP headers which can lead to many opportunities
+  for XSS and CRSF by attackers. After receiving and interpreting a request message, a server responds with an HTTP response message.
+  \0 character inside a headerline is being interpreted as a terminator.
+  But \00 can also work
+  An attacker can bypass wep application filters like this -
```
Comment (msp2248) [inline cves/CVE-2011-3880.yml]: web*

File: cves/CVE-2011-3880.yml [hunk 3]
```diff
@@ -197,4 +209,5 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: This was a coding issue with how a chromium server properly whitelists user data inserted into headers.
```
Comment (msp2248) [inline cves/CVE-2011-3880.yml]: Elaborate a bit more in this section

**Comment by mesternefeld** [comment]
Please remove the gitlog file from the pr.

File: cves/CVE-2011-3880.yml [hunk 4]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: 86 | 44
```
Comment (mesternefeld) [inline cves/CVE-2011-3880.yml]: please only add one CWE

File: cves/CVE-2011-3880.yml [hunk 5]
```diff
@@ -106,8 +117,9 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: The mistake was made in the networksubsystem of chromium. The original file changed for the fix  is ('net/http_utils..).
```
Comment (mesternefeld) [inline cves/CVE-2011-3880.yml]: please make this a multi-line comment
```suggestion
  answer: |
    The mistake was made in the networksubsystem of chromium. The original file changed for the fix  is 
    ('net/http_utils..).
```

File: cves/CVE-2011-3880.yml [hunk 6]
```diff
@@ -58,7 +68,6 @@ vccs:
   :note: 
 upvotes_instructions: |
```
Comment (daviematz) [inline cves/CVE-2011-3880.yml]: +3 upvotes

**Comment by Jcuevas97** [comment]
+4 upvotes

File: cves/CVE-2011-3880.yml [hunk 7]
```diff
@@ -183,7 +196,7 @@ lessons:
     applies: 
     note: 
   complex_inputs:
-    applies: 
+    applies: true
     note:
```
Comment (blackpan2) [inline cves/CVE-2011-3880.yml]: Please fill in how this applies to the vulnerability.

File: cves/CVE-2011-3880.yml [hunk 8]
```diff
@@ -106,8 +117,10 @@ subsystem:
...
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: |
+    The mistake was made in the network subsystem of chromium. The original file changed for the fix  is ('net/http_utils..).
+    In the chromium bug report, the component listed as affected was 'Internals > Network)
+  name: Network
 interesting_commits:
   question: |
     Are there any interesting commits between your VCC(s) and fix(es)?
```
Comment (blackpan2) [inline cves/CVE-2011-3880.yml]: `If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.`

File: cves/CVE-2011-3880.yml [hunk 9]
```diff
@@ -197,4 +210,7 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: This was a coding issue with how a chromium server properly whitelists user data inserted into headers. The current fix, ensures that chrome
+    neutralizes the invalid character (\0) in HTTP headers.for XSS and CRSF by attackers
```
Comment (blackpan2) [inline cves/CVE-2011-3880.yml]: "headers.for" -> "headers for"
"by attackers" -> "by attackers."

File: cves/CVE-2014-7925.yml [hunk 10]
```diff
@@ -65,7 +67,8 @@ fixes:
 - :commit: 323c29638687ab5c5c542c803e87a748baada6bf
   :note: ''
 vccs:
-- :commit:
+- :commit: Checked the git logs, did git blame on the fix commit and referenced the changed http_utils.cc files. I also went through chromiums
+    code review for the fix to see if any record of the VCC was accessible. All were in vain.
   :note: 
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (blackpan2) [inline cves/CVE-2014-7925.yml]: If there were no upvotes for this CVE please list 0 on the appropriate line

File: cves/CVE-2014-7925.yml [hunk 11]
```diff
@@ -105,7 +109,8 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
+  answer: Checked the git logs, did git blame on the fix commits and referenced the changed AudioSummingJunction.cpp file. I also went through chromiums
```
Comment (blackpan2) [inline cves/CVE-2014-7925.yml]: It appears that the issue was created by a community member ([Reported by mark.b...@gmail.com, Nov 18 2014](https://bugs.chromium.org/p/chromium/issues/detail?id=434136)) it contains steps to reproduce it and expected behavior, we can assume that this is when the issue was discovered.

File: cves/CVE-2014-7925.yml [hunk 12]
```diff
@@ -117,8 +122,9 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: The mistake was made in the networksubsystem of chromium. The original file changed for the fix  is ('modules/webaudio/AudioSummingJunction.cpp').
```
Comment (blackpan2) [inline cves/CVE-2014-7925.yml]: "networksubsystem" -> "network subsystem"

File: cves/CVE-2014-7925.yml [hunk 13]
```diff
@@ -208,4 +214,5 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: In my opinion, this was a design issue. The way threads are handled which allow data and processes that are being used by
+    chrome are also being cleared out by the Garbage collector process.
```
Comment (blackpan2) [inline cves/CVE-2014-7925.yml]: "The way threads are handled which allow data and processes that are being used by chrome are also being cleared out by the Garbage collector process"

Reading this out loud sounds odd. Adding some commas groups the thoughts in a more clear way.
"The way threads are handled, which allow data and processes, that are being used by chrome are also being cleared out by the Garbage collector process"

**Pull request merged**